components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt

```diff
@@ -0,0 +1,547 @@
+     */
+    fun optOut(experimentId: String) = Unit
+
+    /**
+     *  Reset internal state in response to application-level telemetry reset.
+     *  Consumers should call this method when the user resets the telemetry state of the
+     *  consuming application, such as by opting out of (or in to) submitting telemetry.
+     */
+    fun resetTelemetryIdentifiers() = Unit
+
+    /**
+     * Control the opt out for all experiments at once. This is likely a user action.
+     */
+    var globalUserParticipation: Boolean
+        get() = false
+        set(_) = Unit
+
+    /**
+     * Interface to be implemented by classes that want to observe experiment updates
+     */
+    interface Observer {
+        /**
+         * Event to indicate that the experiments have been fetched from the endpoint
+         */
+        fun onExperimentsFetched() = Unit
+
+        /**
+         * Event to indicate that the experiment enrollments have been applied. Multiple calls to
+         * get the active experiments will return the same value so this has limited usefulness for
+         * most feature developers
+         */
+        fun onUpdatesApplied(updated: List<EnrolledExperiment>) = Unit
+    }
+}
+
+/**
+ * This class allows client apps to configure Nimbus to point to your own server.
+ * Client app developers should set up their own Nimbus infrastructure, to avoid different
+ * organizations running conflicting experiments or hitting servers with extra network traffic.
+ */
+data class NimbusServerSettings(
+    val url: Uri,
+    val bucket: String = EXPERIMENT_BUCKET_NAME,
+    val collection: String = EXPERIMENT_COLLECTION_NAME
+)
+
+typealias ErrorReporter = (message: String, e: Throwable) -> Unit
+
+private typealias LoggerFunction = (message: String) -> Unit
+
+/**
+ * This class represents the client application name and channel for filtering purposes
+ */
+data class NimbusAppInfo(
+    /**
+     * The app name, used for experiment filtering purposes so that only the intended application
+     * is targeted for the experiment.
+     *
+     * Examples: "fenix", "focus".
+     *
+     * For Mozilla products, this is defined in the telemetry system. For more context on where the
+     * app_name comes for Mozilla products from see:
+     * https://probeinfo.telemetry.mozilla.org/v2/glean/app-listings
+     * and
+     * https://github.com/mozilla/probe-scraper/blob/master/repositories.yaml
+     */
+    val appName: String,
+    /**
+     * The app channel used for experiment filtering purposes, so that only the intended application
+     * channel is targeted for the experiment.
+     *
+     * Examples: "nightly", "beta", "release"
+     */
+    val channel: String
+)
+
+/**
+ * Small struct for info derived from the device itself.
+ */
+data class NimbusDeviceInfo(
+    val localeTag: String
+)
+
+/**
+ * Provide calling apps control how Nimbus fits into it.
+ */
+data class NimbusDelegate(
```

Comment:
nit: Is there a reason this is a data class? You don't need any of the generated functions, and `logger`/`errorReporter` are essentially behaviour passed down.

Reply:
Good spot. This was an error.